CommunicationLayer/MessagingFramework/MessageDecodingHelpers.cpp

```diff
@@ -38,12 +38,29 @@ namespace
 
 float MessageDecodingHelpers::getFloat(const QByteArray& data, int startIndex)
 {
-    return data.mid(startIndex, numberOfBytesInData(Type::FLOAT)).toFloat();
+    float value;
+
+    QDataStream dataStream(data.mid(startIndex, numberOfBytesInData(Type::FLOAT)));
+
+    dataStream.setByteOrder(QDataStream::LittleEndian);
+    dataStream.setFloatingPointPrecision(QDataStream::SinglePrecision);
+
+    dataStream >> value;
+
+    return value;
 }
 
 unsigned short MessageDecodingHelpers::getUnsignedShort(const QByteArray& data, int startIndex)
 {
-    return data.mid(startIndex, numberOfBytesInData(Type::UNSIGNED_SHORT)).toUShort();
+    unsigned short value;
```

Comment:
These should be moved closer to where they're being used.
And I'd say there is more empty lines in these functions then are necessary.